### DISCUSSION TITLE
Drive GPIO NRF51 doesn't work

### DISCUSSION THREAD

**Discussion by ArnaudMartinCaen** [micropython/micropython#12922] [nRF]
Hello,

I am trying to drive PIN on NRF51822 of my microbitv1 to drive a LED with code below :

from machine import Pin

Row1 = Pin(13, Pin.OUT)
Row1.high()

Col1 = Pin(4, Pin.OUT)
Col1.low()

But it doesn't work.

I try to use on() and off() or value() function but i have the same problem.

When i add theses lines :

from microbit import *
display.set_pixel(0,0,1)

It works, but i would like to use a function to drive Pin like on, off or value.

Thanks for your help.

**Comment by jimmo** [comment]
@ArnaudMartinCaen Please see the reply I sent on Discord. Please avoid duplicate-posting (or at least cross-reference your posts) to avoid people accidentally answering a question that's been already answered somewhere else.

(For anyone else who runs into the same thing: the answer is that the microbit firmware runs the display in a background task which is always running. So whatever pin state you set is immediately overridden by that task).
